include/networkit/graph/Graph.hpp

```diff
@@ -590,25 +595,34 @@ class Graph final {
      */
     template <bool InEdges = false> class NeighborRange {
       public:
-        NeighborRange(const Graph &G, node u) : G(G), u(u){};
+        NeighborRange(const Graph &G, node u) : G(&G), u(u){
+			assert(exists[u]);
```

Comment:
This has to be `assert(G.hasNode(u));`

Reply:
Thanks for this - this is the reason the CI was failing. I have changed it, as well as the other issues.